### DISCUSSION TITLE
[feat][question] Is there any way to get parameter in Atom Effect?

### DISCUSSION THREAD

**Discussion by Gumichocopengin8** [facebookexperimental/Recoil#2022] [Q&A] [answered]
Is there any way to get parameter in Atom Effect of atomFamily?

If no, I want to have a feature to get parameter in Atom Effect.

```ts
const someState = atomFamily<string, string>({
    key: "key",
    default: paran => param,
    effect: [
        ({ onSet, getParam, node }) => { // <-- getParam would be nice to have as a new feature
            onSet((newValue) => {
              const param = getParam(node);
              console.log(param)
            })
        }
    ]
})
```

**Comment by wd-fb** [comment, marked as answer]
Hello, and yes!   The family parameter can be obtained by using a callback for `effects` as noted here:

https://recoiljs.org/docs/recoil-sync/sync-effect/#atom-families

see also: https://recoiljs.org/docs/guides/atom-effects/

**Reply by Gumichocopengin8** [reply]
Hello, 
Thank you for the information. https://recoiljs.org/docs/recoil-sync/sync-effect/#atom-families example is very helpful.
I first looked into Atom Effect doc but it was a bit confusing about param
